dropwizard-testing/src/main/java/io/dropwizard/testing/common/DropwizardTestResourceConfig.java

```diff
@@ -48,6 +64,27 @@
         for (Supplier<?> singleton : configuration.singletons) {
             register(singleton.get());
         }
+        for (Supplier<?> resource : configuration.resources) {
```

Comment:
without separating out Resources from Provider singletons, dropwizard-example's `Protected*ResourceTest`s fail trying to bind/register a `Class`